naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/SangriaGraphQlSchemaBuilder.scala

```diff
@@ -56,219 +39,39 @@ class SangriaGraphQlSchemaBuilder(
 
   /**
     * Generates a GraphQL schema for the provided set of resources to this class
-    * Returns a "root" object that has one field available for each Naptime Resource provided.*
+    * Returns a "root" object that has one field available for each Naptime Resource provided.
     *
-    * @return a Sangria GraphQL Schema with all resources defined
+    * @return a Sangria GraphQL Schema with all resources defined,
+    *         and a list of errors that came up while generating the schema
     */
-  def generateSchema(): Schema[SangriaGraphQlContext, DataMap] = {
-    val topLevelResourceObjects = for {
-      resource <- resources
-      resourceObject <- (try {
-        val resourceName = ResourceName(
-          resource.name, resource.version.getOrElse(0L).toInt).identifier
-        generateLookupTypeForResource(resourceName)
-      } catch {
-        case e: Throwable => None
-      }).toList if resourceObject.fields.nonEmpty
-    } yield {
-      Field.apply[SangriaGraphQlContext, DataMap, DataMap, Any](
-        formatResourceTopLevelName(resource),
-        resourceObject,
-        resolve = (context: Context[SangriaGraphQlContext, DataMap]) => {
-          Value(new DataMap())
-        })
+  def generateSchema(): WithSchemaErrors[Schema[SangriaGraphQlContext, DataMap]] = {
+    val topLevelResourceObjectsAndErrors = resources.map { resource =>
+      val lookupTypeAndErrors =
+        NaptimeTopLevelResourceField.generateLookupTypeForResource(resource, schemaMetadata)
```

Comment:
For my own learning, I'd like to know why we favor static objects over injected dependencies. From my own experience, I've found that it's a lot easier to test when dependencies can be mocked out and static objects should only be used when absolutely necessary. 

i.e. since I have no idea how deep the call graph goes on these static objects, I'd like to use `when` matchers wherever possible.